tick/optim/model/src/poisreg.cpp

```diff
@@ -71,12 +76,101 @@ double ModelPoisReg::sdca_dual_min_i(const ulong i,
   // Check we are in the correct bounds
   if (new_dual >= label) {
     new_dual = label - epsilon;
-    delta_dual = new_dual - dual;
+    delta_dual = new_dual - dual_i;
   }
 
   return delta_dual;
 }
 
+void ModelPoisReg::init_non_zero_label_map() {
+  non_zero_labels = VArrayULong::new_ptr();
+  for (ulong i = 0; i < get_n_samples(); ++i) {
+    if (get_label(i) != 0) {
+      non_zero_labels->append1(i);
+    }
+  }
+  n_non_zeros_labels = non_zero_labels->size();
+  non_zero_label_map_computed = true;
+}
+
+double ModelPoisReg::sdca_dual_min_i_identity(const ulong i,
+                                              const double dual_i,
+                                              const ArrayDouble &primal_vector,
+                                              const double previous_delta_dual_i,
+                                              double l_l2sq) {
+
+  if (fit_intercept) {
+    TICK_ERROR("SDCA not implemented with fit intercept yet");
```

Comment:
Alright, I thought it was failing...

It works straight forward with identity link so I will just remove this line
```
import numpy as np
from tick.simulation import SimuPoisReg, weights_sparse_gauss
from tick.plot import stems
from tick.optim.model import ModelPoisReg
from tick.optim.solver import SDCA
from tick.optim.prox import ProxZero, ProxL1, ProxL2Sq

import matplotlib.pyplot as plt
%matplotlib inline

n_samples = 30000
n_features = 10

weight0 = np.random.rand(n_features)
features = np.random.rand(n_samples, n_features)

X, y = SimuPoisReg(weight0, intercept0, n_samples=n_samples, features=features, link='identity').simulate()

model = ModelPoisReg(fit_intercept=True, link='identity')
model.fit(X, y)
prox = ProxZero()

solver = SDCA(1e-3, tol=1e-10, max_iter=1000, print_every=100)
solver.set_model(model)
solver.set_prox(prox)
solver.solve()

plt.figure(figsize=(6, 3))
plt.subplot(1, 2, 1)
plt.stem(np.concatenate([weight0, [intercept0]]))
plt.title('ground truth')
plt.subplot(1, 2, 2)
plt.stem(solver.solution)
plt.title('sdca')
```